File: sql_ts/src/orders/getOrder.ts
import { DynamoDB } from 'aws-sdk';
import { HttpRequest, HttpResponse } from '../http'

const ddb = new DynamoDB.DocumentClient({ region: "eu-central-1" })

export async function handler(event: HttpRequest): Promise<HttpResponse> {
  const userId = event.pathParameters['userId']
  const params = {
    TableName: process.env.ORDERS_TABLE!,
    Key: {
      userId: userId,
      orderNo: event.pathParameters.orderNo
    },
  }

  try {
    const data = await ddb.get(params).promise()
    if (data.Item) {
      const order = data.Item
      return {
        statusCode: 200,
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Credentials': true,
        },
        body: JSON.stringify(order)
      }
    }
    return {
      statusCode: 404,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: "Item not found"
    }
  } catch (err) {
    console.error(err)
    return {
      statusCode: err.statusCode || 500,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: "Internal Server Error Oops" + err
    }
  }
}

File: dynamo_go/lambda/orders/get/getOrder.go
package main

import (
	"encoding/json"
	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/dynamodb"
	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
	"os"
	"webshop/internal/apigw"
	"webshop/internal/forms"
)

func Handler(event events.APIGatewayProxyRequest) (*events.APIGatewayProxyResponse, error) {
	ddb := dynamodb.New(session.New())

	out, err := ddb.GetItem(&dynamodb.GetItemInput{
		TableName: aws.String(os.Getenv("ORDERS_TABLE")), //aws.String macht wert zu einem  pointer
		Key: map[string]*dynamodb.AttributeValue{
			"orderNo": {S: aws.String(event.PathParameters["orderNo"])},
			"userId":  {S: aws.String(event.PathParameters["userId"])},
		},
	})
	if err != nil {
		return apigw.CreateResponse(500, err.Error()), err
	}

	if len(out.Item) > 0 {
		var order forms.Order
		if err := dynamodbattribute.UnmarshalMap(out.Item, &order); err != nil {
			return apigw.CreateResponse(500, err.Error()), err
		}
		jsonStr, err := json.Marshal(order)
		if err != nil {
			return apigw.CreateResponse(500, err.Error()), err
		}
		return apigw.CreateResponse(200, string(jsonStr)), err
	}
	return apigw.CreateResponse(404, "Not found"), err
}

func main() {
	lambda.Start(Handler)
}
